pkg/types/azure/validation/disk.go

```diff
@@ -31,7 +31,7 @@ func ValidateDiskEncryption(p *azure.MachinePool, cloudName azure.CloudEnvironme
 	if diskEncryptionSet != nil && cloudName == azure.StackCloud {
 		return append(allErrs, field.Invalid(childFldPath.Child("diskEncryptionSet"), diskEncryptionSet, "disk encryption sets are not supported on this platform"))
 	}
-	if diskEncryptionSet.SubscriptionID != "" && !RxSubscriptionID.MatchString(diskEncryptionSet.SubscriptionID) {
+	if !RxSubscriptionID.MatchString(diskEncryptionSet.SubscriptionID) {
```

Comment:
nit: `field.Required` would be a more appropriate error when `diskEncryptionSet.SubscriptionID == ""`.

Reply:
Fixed.